phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -311,8 +313,10 @@
                     IMMUTABLE_STORAGE_SCHEME + "," +
                     ENCODING_SCHEME + "," +
                     USE_STATS_FOR_PARALLELIZATION +"," +
-                    VIEW_INDEX_ID_DATA_TYPE +
-                    ") VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)";
+                    VIEW_INDEX_ID_DATA_TYPE +"," +
```

Comment:
I don't understand the logic of adding a new column to SYSTEM.CATALOG table. for the server-side update, do we drop the table and re-create a table?

Reply:
The view-ttl properties will be stored/persisted as part of the view header columns.

Reply:
I was asking how to upgrade client and server side changes because it requires a new release or tool to doing this.

Reply:
As I mentioned before, we will need to add steps in the upgrade path `CQSI.upgradeSystemTables` to ensure that new columns are added to syscat when a client upgrades from 4.x metadata to 4.(x+1), assuming the VIEW_TTL feature goes into 4.(x+1)

Reply:
OR, we will need to create a new tool like the IndexUpgradeTool to do the same syscat table alteration in order to enable using this feature for a patch upgrade i.e. 4.x.y -> 4.x.(y+1)